### PR TITLE
Chatbot_Johanna Rosenholm and Maria So

### PR DESCRIPTION
**Pull request by RosenholmJohanna**

### CODE REVIEW

**Comment by RosenholmJohanna** [comment]
https://sparkly-cannoli-70d42e.netlify.app/

**Comment by Iracuzzi** [comment]
Overall really good structure of the code, easy to follow and understand. 
The naming of the different functions is really good and easily to understand/follow.
We can see that you are using setTimout to delay the answer from bot/user but not for all bot answer. It can be an idea to add it for those bot answer where it's missing in the code to make an even better flow in the chat. See comments in code.
After all It is a relly good work, and the chatbot work really fine!

File: code/script.js [hunk 1]
```diff
@@ -1,51 +1,217 @@
...
-const chat = document.getElementById('chat');
+// Variables that point to selected DOM elements  //declaring these elements from html
+const chat = document.getElementById('chat')                  //this the whole chatbox area
+const inputWrapper = document.getElementById('inputWrapper')   //the whole chat-box
+const nameForm = document.getElementById('nameForm')          //input field
+const nameInput = document.getElementById ('nameInput')      //typed input
+const button = document.getElementById('button')            //button
 
-// If you need any global variables that you can use across different functions, declare them here:
+
```
Comment (Iracuzzi) [inline code/script.js:8]: We suggest to make more easy to code the answers in the chat, do a function for the bot replay and the user replay, like that:
`const userReplay = (message) => {
    showMessage(message, 'user')
const botReplay = (message) => {
    showMessage(message, 'bot')`

File: code/script.js [hunk 2]
```diff
@@ -1,51 +1,217 @@
...
-const greeting = () => {
-  // here we call the function showMessage, that we declared earlier with the argument "Hello there, What's your name?" for message, and the argument "bot" for sender
-  showMessage("Hello there, What's your name?", 'bot');
-  // Just to check it out, change 'bot' to 'user' here 👆
+const greetingAnswer = (event) => { 
+  event.preventDefault();   //prevent website refresh at submit 
+  const answer = nameInput.value //storing the answer in a variable
+  showMessage(`My name is ${answer}`, 'user');   //the greeting answer with name from user 
+  nameInput.value = ''          //setting the users answer into an empty string
+  showMessage(`Hello! ${answer} Choose a topic you would like to know about?` ,  'bot');
```
Comment (Iracuzzi) [inline code/script.js:55]: Here we think you miss one setTimeout.

File: code/script.js [hunk 3]
```diff
@@ -1,51 +1,217 @@
...
+    setTimeout(() => answerTwo('romance' , 'user')),1000  
+    }) 
+    document.getElementById('nature').addEventListener('click' , () => { showMessage("Nature is my favourite choice!", 'user')
+    setTimeout(() => answerTwo('nature' , 'user')),1000  
+    }) 
+    document.getElementById('unicorns').addEventListener('click' , () => { showMessage("Give me the magic, picking unicorns ofc!", 'user')
+    setTimeout(() => answerTwo('unicorns' , 'user')),1000  
+    }) 
+    document.getElementById('food').addEventListener('click' , () => { showMessage("I'm excited to know about food.", 'user')
+    setTimeout(() => answerTwo('food' , 'user')),1000
```
Comment (Iracuzzi) [inline code/script.js:81]: In this lines you are missing the closing parenthesis in all the setTimeout.

File: code/script.js [hunk 4]
```diff
@@ -1,51 +1,217 @@
...
+        setTimeout(() => showMessage("Ah, romance my favorite choice! Choose your category.", 'bot'),1000)
+        inputWrapper.innerHTML = `
+        <button id = "single" type = "submit">single</button>
+        <button id = "marriage" type = "submit">marriage</button>
+        `
+        document.getElementById('single').addEventListener('click' , () => { showMessage("Dying to know about singleness.", 'user')
+        setTimeout(() => facts('single' , 'user')),1000  // calling out the next answer in a new function named facts
+        }) 
+        document.getElementById('marriage').addEventListener('click' , () => { showMessage("Married facts for married people.", 'user')
+        setTimeout(() => facts('marriage' , 'user')),1000  // calling out the next answer in a new function named facts
```
Comment (Iracuzzi) [inline code/script.js:97]: In this lines you are missing the closing parenthesis in all the setTimeout.

File: code/script.js [hunk 5]
```diff
@@ -1,51 +1,217 @@
...
+        setTimeout(() => showMessage("Let's give you some nature wisdom, category?", 'bot'),1000)
+        inputWrapper.innerHTML = `
+        <button id = "ocean" type = "submit">ocean</button>
+        <button id = "animals" type = "submit">animals</button>
+        `
+        document.getElementById('ocean').addEventListener('click' , () => { showMessage("Tell me about this, please", 'user')
+        setTimeout(() => facts('ocean' , 'user')),1000  // calling out the next answer in a new function named facts
+        }) 
+        document.getElementById('animals').addEventListener('click' , () => { showMessage("Animals, are my fave! Give me the info!", 'user')
+        setTimeout(() => facts('animals' , 'user')),1000  // calling out the next answer in a new function named facts
```
Comment (Iracuzzi) [inline code/script.js:112]: In this lines you are missing the closing parenthesis in all the setTimeout.

File: code/script.js [hunk 6]
```diff
@@ -1,51 +1,217 @@
...
+        inputWrapper.innerHTML = `
+        <button id = "superpower" type = "submit">superpower</button>
+        <button id = "wings" type = "submit">wings</button>
+        `
+        document.getElementById('superpower').addEventListener('click' , () => { showMessage("Tell me something about their superpower!", 'user')
+        setTimeout(() => facts('superpower' , 'user')),1000  // calling out the next answer in a new function named facts
+        }) 
+
+        document.getElementById('wings').addEventListener('click' , () => { showMessage("Tell me something about their wings!", 'user')
+        setTimeout(() => facts('wings' , 'user')),1000  // calling out the next answer in a new function named facts
```
Comment (Iracuzzi) [inline code/script.js:130]: In this lines you are missing the closing parenthesis in all the setTimeout.

File: code/script.js [hunk 7]
```diff
@@ -1,51 +1,217 @@
...
+        inputWrapper.innerHTML = `
+        <button id = "fastfood" type = "submit">fastfood</button>
+        <button id = "health" type = "submit">health themed</button>
+        `
+       
+        document.getElementById('fastfood').addEventListener('click' , () => { showMessage("Tell me about fastfood, please.", 'user')
+        setTimeout(() => facts('fastfood' , 'user')),1000  // calling out the next answer in a new function named facts
+        }) 
+        document.getElementById('health').addEventListener('click' , () => { showMessage("Health info for the healthbuff!", 'user')
+        setTimeout(() => facts('health' , 'user')),1000  // calling out the next answer in a new function named facts
```
Comment (Iracuzzi) [inline code/script.js:146]: In this lines you are missing the closing parenthesis in all the setTimeout.

File: code/script.js [hunk 8]
```diff
@@ -1,51 +1,217 @@
...
+            setTimeout(() => showMessage("I will tell you the truth: An ostrich's eyes are bigger than its brain!", 'bot'),1000)   
+            setTimeout(() => closure('single', 'bot')),1000  // calling out the next answer in a new function named CLOSURE
+        }
+        else if (lastChoice === 'fastfood'){
+            setTimeout(() => showMessage("Fact: The world's largest toy distributor is none other than McDonald's, thanks to their Happy Meals", 'bot'),1000)  
+            setTimeout(() => closure('single', 'bot')),1000  // calling out the next answer in a new function named CLOSURE 
+        }
+        else if (lastChoice === 'health'){
+            setTimeout(() => showMessage("Fact: There Is No Perfect Diet for Everyone.", 'bot'),1000)   
+            setTimeout(() => closure('single', 'bot')),1000  // calling out the next answer in a new function named CLOSURE
```
Comment (Iracuzzi) [inline code/script.js:185]: In this lines you are missing the closing parenthesis in all the setTimeout.

File: code/script.js [hunk 9]
```diff
@@ -1,51 +1,217 @@
...
+
+        inputWrapper.innerHTML = `
+        <button id = "yes" type = "submit">Yes, more facts.</button>
+        <button id = "no" type = "submit">End chat.</button>
+        `
+        document.getElementById('yes').addEventListener('click' , () => { showMessage("I want to know more!", 'user')
+        setTimeout(() => questionTwo('yes' , 'user')),1000  // calling out the next answer in a new function named facts
+        })
+        document.getElementById('no').addEventListener('click' , () => { showMessage("No thank you, had fun though!", 'user')
+        setTimeout(() => bye('no' , 'user')),1000  // calling out the next answer in a new function named facts
```
Comment (Iracuzzi) [inline code/script.js:202]: In this lines you are missing the closing parenthesis in all the setTimeout.

**Review by Iracuzzi — COMMENTED** [review]
Overall really good structure of the code, easy to follow and understand.
The naming of the different functions is really good and easily to understand/follow.
We can see that you are using setTimout to delay the anser from bot/user. What we notice is that closing parantes is missing after some of setTimeout for the answers that needs to be added and we suggest to add an setTimout for the first bot reply to make an even better flow. See comments
After all It is a relly good work, and the chatbot work really fine!

**Pull request closed without merging**